### ISSUE TITLE
segfault after long run

### ISSUE THREAD

**Issue by onlyjob** [tgraf/bmon#31]
In https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=798995 user reported segfault after leaving `bmon` running for a day. Unfortunately his backtrace is not great...

**Comment by tgraf** [comment]
I've pushed commit f9c714fa1503179896c55ccb5f918e6c5d6cd5f5 which might fix this. I couldn't reproduce the corruption itself but found another related issue which could cause this as well.

**Comment by OmegaPhil** [comment]
I have compiled and installed the latest git and will see how it fares.

**Comment by OmegaPhil** [comment]
Nope, segfaulted at some point in the last 17 hours. 99% probability of the same bug, is it worth getting a backtrace out of it?
